src/node.h

```diff
@@ -612,9 +612,20 @@ NODE_EXTERN void AtExit(void (*cb)(void* arg), void* arg = 0);
  */
 NODE_EXTERN void AtExit(Environment* env, void (*cb)(void* arg), void* arg = 0);
 
-typedef void (*promise_hook_func) (v8::PromiseHookType type,
+NODE_EXTERN typedef enum {
```

Comment:
I don’t think it really makes sense to add `NODE_EXTERN` to a type definition; also, the `typedef` is redundant, just using `enum PromiseHookType { … };` should work the same way

Reply:
this doesn't compile without the `typedef`